### ISSUE TITLE
uv_interface_addresses on Windows returns disconnected interfaces as well

### ISSUE THREAD

**Issue by gyohng** [joyent/libuv#453]
It seems that on Linux/Unix the interface state is actually checked against IFF_UP / IFF_RUNNING, while on Windows the function returns all (including disconnected) interfaces.

The simplest solution would be to check OperStatus field of the winows structure. Please check the provided patch.

More advanced idea is to extend uv_interface_address_t type with a field to denote interface state.

Thanks,
George.

Patch:

``` diff
--- util.c  2012-05-29 10:08:24.000000000 +0200
+++ util.c  2012-06-08 20:06:09.692858800 +0200
@@ -616,8 +616,13 @@
   for (adapter_address = adapter_addresses;
        adapter_address != NULL;
        adapter_address = adapter_address->Next) {
+
+    if (adapter_address->OperStatus!=IfOperStatusUp) 
+      continue;
+
     unicast_address = (IP_ADAPTER_UNICAST_ADDRESS_XP*)
                       adapter_address->FirstUnicastAddress;
+
     while (unicast_address) {
       (*count)++;
       unicast_address = unicast_address->Next;
@@ -635,6 +640,10 @@
   for (adapter_address = adapter_addresses;
        adapter_address != NULL;
        adapter_address = adapter_address->Next) {
+
+    if (adapter_address->OperStatus!=IfOperStatusUp) 
+      continue;
+
     name = NULL;
     unicast_address = (IP_ADAPTER_UNICAST_ADDRESS_XP*)
                       adapter_address->FirstUnicastAddress;
```

**Comment by gyohng** [comment]
Sorry, I have no idea how to use github commenting. Here is a link to the patch file:

http://www.yohng.com/pvt/src_win_util.patch

**Comment by piscisaureus** [comment]
@gyohng Can you sign the node.js CLA? (http://nodejs.org/cla.html) Then I can land this patch.

**Comment by gyohng** [comment]
Done.

**Comment by piscisaureus** [comment]
@gyohng Thanks, landed in fbe99743d5889a72112e1ce33af9e490bd39a677

Btw, if you ever do a patch again, we prefer using github pull requests, or git-style patch emails (created by `git --format-patch`).
